packages/rapidpro/src/Adaptor.js

```
@@ -104,41 +103,45 @@ export function upsertContact(params, callback) {
       headers: { Authorization: `Token ${token}` },
     };
 
-    return http
-      .post(config)(state)
-      .then(resp => {
-        console.log('Contact added with uuid:', resp.data.uuid);
-        return resp;
-      })
-      .catch(err => {
-        const { data } = err.response;
-        if (
-          data &&
-          data.urns &&
-          Array.isArray(data.urns) &&
-          data.urns.find(x => x.includes('URN belongs to another'))
-        ) {
+    return await fetch(url, {
+      method: 'POST',
+      headers: {
+        Authorization: `Token ${token}`,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(config.data),
+    })
+      .then(async response => {
+        const result = await response.json();
+        if (result && result.urns && Array.isArray(result.urns['0'])) {
           const newUrl = `${url}?urn=${config.data.urns[0]}`;
-          delete config.data['urns'];
-          return http
-            .post({ ...config, url: newUrl })(state)
-            .then(resp => {
-              console.log('Contact updated with uuid:', resp.data.uuid);
-              return resp;
-            });
-        } else {
-          console.log(JSON.stringify(data, null, 2));
 
-          delete err.response.request;
-          delete err.response.config;
+          delete config.data['urns'];
+          return await fetch(newUrl, {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json',
+              ...config.headers,
+            },
+            body: JSON.stringify(config.data),
+          }).then(async response => {
+            const res = await response.json();
 
-          throw err.response;
+            console.log('Contact updated with uuid:', res.uuid);
+            return res;
+          });
+        } else {
+          console.log('Contact added with uuid:', result.uuid);
+          return result;
         }
       })
+      .catch(async err => {
```

You don't need to keep this catch - just let the promise throw